src/Tinc/Freeze.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Tinc.Freeze (
+  writeFreezeFile
+, readFreezeFile
+) where
+
+import           Data.Aeson
+import qualified Data.ByteString as B
+import           Data.Char
+import           Data.Ord
+import           Data.List
+import qualified Data.Yaml as Yaml
+import           Hpack.Yaml
+import           GHC.Generics
+import           System.Directory
+
+import           Tinc.Fail
+import           Tinc.Package
+
+type Constrain = String
+
+freezeFile :: FilePath
+freezeFile = ".tinc.freeze"
+
+data Dependency = Dependency {
+  name :: String
+, version :: String
+} deriving (Eq, Show, Generic)
+
+instance FromJSON Dependency
+
+instance ToJSON Dependency where
+  toJSON = genericToJSON defaultOptions
```

Comment:
With `aeson-0.10.0.0` you get a warning if you don't specify it explicitly... :(:(:(